src/lerobot/datasets/v30/convert_dataset_v30_to_v21.py

```diff
@@ -0,0 +1,509 @@
+
+            records = sorted(records, key=lambda rec: float(rec[f"videos/{video_key}/from_timestamp"]))
+
+            for record in records:
+                episode_index = int(record["episode_index"])
+                start = float(record[f"videos/{video_key}/from_timestamp"])
+                end = float(record[f"videos/{video_key}/to_timestamp"])
+
+                dest_chunk = episode_index // DEFAULT_CHUNK_SIZE
+                dest_path = new_root / LEGACY_VIDEO_PATH_TEMPLATE.format(
+                    chunk_index=dest_chunk,
+                    video_key=video_key,
+                    episode_index=episode_index,
+                )
+
+                _extract_video_segment(src_path, dest_path, start=start, end=end)
+
+
+def convert_episodes_metadata(new_root: Path, episode_records: list[dict[str, Any]]) -> None:
+    logging.info("Reconstructing legacy episodes and episodes_stats JSONL files")
+
+    episodes_path = new_root / LEGACY_EPISODES_PATH
+    stats_path = new_root / LEGACY_EPISODES_STATS_PATH
+    episodes_path.parent.mkdir(parents=True, exist_ok=True)
+
+    with jsonlines.open(episodes_path, mode="w") as episodes_writer, jsonlines.open(
+        stats_path, mode="w"
+    ) as stats_writer:
+        for record in sorted(episode_records, key=lambda rec: int(rec["episode_index"])):
+            legacy_episode = {
+                key: value
+                for key, value in record.items()
+                if not key.startswith("data/")
+                and not key.startswith("videos/")
+                and not key.startswith("stats/")
+                and not key.startswith("meta/")
+                and key not in {"dataset_from_index", "dataset_to_index"}
+            }
+
+            serializable_episode = {key: _to_serializable(value) for key, value in legacy_episode.items()}
+            episodes_writer.write(serializable_episode)
+
+            stats_flat = {key: record[key] for key in record if key.startswith("stats/")}
+            stats_nested = unflatten_dict(stats_flat).get("stats", {})
+            stats_serialized = serialize_dict(stats_nested)
+            stats_writer.write(
+                {
+                    "episode_index": int(record["episode_index"]),
+                    "stats": stats_serialized,
+                }
+            )
+
+
+def copy_global_stats(root: Path, new_root: Path) -> None:
+    source_stats = root / "meta" / "stats.json"
+    if source_stats.exists():
+        target_stats = new_root / "meta" / "stats.json"
+        target_stats.parent.mkdir(parents=True, exist_ok=True)
+        shutil.copy2(source_stats, target_stats)
+
+
+def copy_ancillary_directories(root: Path, new_root: Path) -> None:
+    for subdir in ["images"]:
+        source = root / subdir
+        if source.exists():
+            shutil.copytree(source, new_root / subdir, dirs_exist_ok=True)
+
+
+def convert_dataset(
+    repo_id: str,
```

Comment:
Thanks for the converstion script, this will be really helpful as currently Nvidia Groot N1.5 expects the LeRobot format to be v2.1. However, can you also add support to convert the data sets locally from V3.0 to V2.1 instead of only supporting the datasets uploaded to Hugging Face?